nautobot/dcim/models/power.py

```diff
@@ -137,30 +174,84 @@ class PowerFeed(PrimaryModel, PathEndpoint, CableTermination):
+                raise ValidationError(
+                    {
+                        "rack": f'Rack "{self.rack}" ({self.rack.location}) and '  # pylint: disable=no-member
+                        f'power panel "{self.power_panel}" ({self.power_panel.location}) '  # pylint: disable=no-member
+                        f"are not in the same location hierarchy."
+                    }
+                )
 
         # AC voltage cannot be negative
         if self.voltage < 0 and self.supply == PowerFeedSupplyChoices.SUPPLY_AC:
             raise ValidationError({"voltage": "Voltage cannot be negative for AC supply"})
 
+        # Destination panel validation
+        if self.destination_panel:
+            # Cannot feed into the same panel
+            if self.destination_panel == self.power_panel:
+                raise ValidationError({"destination_panel": "A power feed cannot connect a panel to itself"})
+
+        # Breaker position and pole validation
+        if self.breaker_position is not None:
+            # Default to single pole breaker when breaker_position is specified but breaker_poles are not
+            if self.breaker_poles is None:
+                self.breaker_poles = PowerFeedBreakerPoleChoices.POLE_1
+
+            # Ensure breaker positions fit within panel capacity
+            if self.power_panel.circuit_positions is not None:
+                occupied_positions = self.get_occupied_positions()
+                if occupied_positions:
+                    max_occupied_position = max(occupied_positions)
+                    if max_occupied_position > self.power_panel.circuit_positions:
+                        raise ValidationError(
+                            {
+                                "breaker_position": f"Breaker configuration starting at position {self.breaker_position} "
+                                f"with {self.breaker_poles} poles would occupy positions {sorted(occupied_positions)}, "
+                                f"but panel only has {self.power_panel.circuit_positions} circuit positions"
+                            }
+                        )
+
+        # Check for breaker position conflicts with other feeds
+        if self.breaker_position is not None and self.breaker_poles is not None:
+            new_positions = self.get_occupied_positions()
+            conflicts = PowerFeed.objects.filter(
+                power_panel=self.power_panel, breaker_position__isnull=False, breaker_poles__isnull=False
+            ).exclude(pk=self.pk if self.pk else None)
+
+            for feed in conflicts:
+                if new_positions.intersection(feed.get_occupied_positions()):
+                    raise ValidationError(
+                        {
+                            "breaker_position": f"Breaker position {self.breaker_position} conflicts with "
+                            f'feed "{feed.name}" (occupies {feed.occupied_positions})'
+                        }
+                    )
```

Comment:
Feels like there's probably a more efficient way to do this, but probably a non-issue if we can generally assume that a given PowerPanel only has < 100 breaker positions?